bridge-adapters/kinetic-bridgehub-adapter-generic-rest/src/main/java/com/kineticdata/bridgehub/adapter/generic/rest/GenericRestAdapter.java

```diff
@@ -393,8 +401,35 @@ protected JSONArray parseResults(Object result, String root) throws BridgeError{
                     DocumentContext jsonContext = JsonPath.parse(jsonObj);
                     jsonRootObj = jsonContext.read(root);
                 } catch (JsonPathException e) {
-                    throw new BridgeError("An issue occured when applying JSON path."
-                        + " Please check the root path in the qualification mapping.", e);
+                    // If contentType is XML, do additional checks since empty XML structures will break JSONPath
+                    if ("XML".equals(contentType)) {
```

Comment:
Would it be possible to move this out of the catch and perform the if in the try block?

Reply:
No, because this only needs to happen if the first try block fails. In XML, if the list you're fetching is empty, the JSONPath root accessor will fail because the child item you want doesn't exist in the XML. So if that happens (the first try block) then I remove the last leaf of the root accessor and check if that exists (the second try block). If it does, then that means the list is empty. If it doesn't then that means the JSONPath root accessor is incorrect and we should throw an error.

Reply:
Ok sounds good.